shell/platform/common/client_wrapper/standard_codec.cc

```diff
@@ -295,13 +295,13 @@ const StandardMessageCodec& StandardMessageCodec::GetInstance(
   if (!serializer) {
     serializer = &StandardCodecSerializer::GetInstance();
   }
-  auto* sInstances = new std::map<const StandardCodecSerializer*,
-                                  std::unique_ptr<StandardMessageCodec>>;
-  auto it = sInstances->find(serializer);
-  if (it == sInstances->end()) {
+  static auto sInstances = std::map<const StandardCodecSerializer*,
```

Comment:
The missing `static` is obviously a mistake, but the other changes you've made here are breading deliberate behavior and violate style: https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables

Reply:
That destructor ordering argument makes sense, thanks for the link. Fixed and added tests. Also did the same in `StandardMethodCodec`, as that's leaking across tests as well.